File: 1803-average-waiting-time/average-waiting-time.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        ans = 0
        t = 0

        for i,j in customers:
            if t<i:
                t = i
            t += j
            ans += t - i
        return ans/len(customers)

File: 1803-average-waiting-time/average-waiting-time.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        double ans = 0;
        int t = 0;
        for(int[] i : customers){
            if(t<i[0]) t = i[0];
            t = t + i[1];
            ans += t - i[0];
        }   
        return ans/customers.length;
    }

}